File: src/main/java/cabalgatas_salento/reservas/controller/AuthController.java
package cabalgatas_salento.reservas.controller;

import cabalgatas_salento.reservas.dto.request.LoginRequest;
import cabalgatas_salento.reservas.dto.request.RegistroRequest;
import cabalgatas_salento.reservas.dto.response.LoginResponse;
import cabalgatas_salento.reservas.service.AuthService;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/auth")
@RequiredArgsConstructor
public class AuthController {

    private final AuthService authService;

    @PostMapping("/login")
    public ResponseEntity<LoginResponse> login(@Valid @RequestBody LoginRequest request) {
        return ResponseEntity.ok(authService.login(request));
    }

    @PostMapping("/registro")
    public ResponseEntity<LoginResponse> registro(@Valid @RequestBody RegistroRequest request) {
        return ResponseEntity.status(HttpStatus.CREATED).body(authService.registro(request));
    }

    /** Solo funciona cuando no existe ningún admin en el sistema. */
    @PostMapping("/registro-admin")
    public ResponseEntity<LoginResponse> registroAdmin(@Valid @RequestBody RegistroRequest request) {
        return ResponseEntity.status(HttpStatus.CREATED).body(authService.registroAdmin(request));
    }
}

File: src/auth/auth.controller.ts
import { Body, Controller, HttpCode, HttpStatus, Post } from '@nestjs/common';
import { Throttle } from '@nestjs/throttler';
import { AuthService } from './auth.service';
import { LoginDto } from './dto/login.dto';
import { LoginResponseDto } from './dto/login-response.dto';
import { RegistroDto } from './dto/registro.dto';
import { Public } from './decorators/public.decorator';

const AUTH_THROTTLE = { default: { ttl: 60000, limit: 5 } };

@Public()
@Controller('auth')
export class AuthController {
  constructor(private readonly authService: AuthService) {}

  @Throttle(AUTH_THROTTLE)
  @Post('login')
  @HttpCode(HttpStatus.OK)
  login(@Body() dto: LoginDto): Promise<LoginResponseDto> {
    return this.authService.login(dto);
  }

  @Post('registro')
  registro(@Body() dto: RegistroDto): Promise<LoginResponseDto> {
    return this.authService.registro(dto);
  }

  @Throttle(AUTH_THROTTLE)
  @Post('registro-admin')
  registroAdmin(@Body() dto: RegistroDto): Promise<LoginResponseDto> {
    return this.authService.registroAdmin(dto);
  }
}
